Source/WebCore/platform/audio/gstreamer/AudioDecoderGStreamer.cpp

```diff
@@ -290,22 +292,21 @@ GStreamerInternalAudioDecoder::GStreamerInternalAudioDecoder(const String& codec
     });
 }
 
-void GStreamerInternalAudioDecoder::decode(std::span<const uint8_t> frameData, bool isKeyFrame, int64_t timestamp, std::optional<uint64_t> duration, AudioDecoder::DecodeCallback&& callback)
+Ref<AudioDecoder::DecodePromise> GStreamerInternalAudioDecoder::decode(std::span<const uint8_t> frameData, bool isKeyFrame, int64_t timestamp, std::optional<uint64_t> duration)
 {
     GST_DEBUG_OBJECT(m_harness->element(), "Decoding%s frame", isKeyFrame ? " key" : "");
 
     auto encodedData = wrapSpanData(frameData);
     if (!encodedData) {
-        m_postTaskCallback([weakThis = WeakPtr { *this }, this, callback = WTFMove(callback)]() mutable {
+        m_postTaskCallback([weakThis = WeakPtr { *this }, this]( {
             if (!weakThis)
                 return;
             if (m_isClosed)
                 return;
 
             m_outputCallback(makeUnexpected("Empty frame"_s));
-            callback({ });
         });
-        return;
+        return AudioDecoder::DecodePromise::reject("Empty frame"_s);
```

Comment:
That will be changing the order of event, promise should be rejected in the lambda, in place of the callback()

Reply:
My hope is to remove entirely postTask.
I don't think this is changing the order of events:
1. postTask is doing: hop to context thread and enqueue a task to execute the callback,
2. resolve will hop to context thread in enqueueTaskWhenSettled via nativePromiseDispatcher, and then will enqueue a task to execute the callback.